### ISSUE TITLE
Publish instructions for how to redeploy a batch

### ISSUE THREAD

**Issue by samreid** [phetsims/decaf#125]
Today we agreed that it would be nice to deploy new translations once a month or so.  So as part of #124 we would like to write up instructions to make it easy for someone else to republish a batch.

**Comment by samreid** [comment]
For #124 I followed the directions in https://github.com/phetsims/decaf/blob/master/README.md, making revisions where necessary.  The first time someone else runs the process it would be most effective to pair with me, so that we can discuss various failure points, how to test, what it looks like when it is working, etc.

@ariel-phet who else on the team would you like to train on this process?

**Comment by ariel-phet** [comment]
Marking for dev meeting to solicit a volunteer

**Comment by jessegreenberg** [comment]
During developer meeting 10/8/20 @Denz1994 said he was happy to help with this. He will create a calendar reminder for early november to see if there are any new translations and it is time to run this process again.

**Comment by samreid** [comment]
@Denz1994 as you follow the instructions for set-up, please revise and improve them as you see deficiencies or problems.

**Comment by Denz1994** [comment]
@samreid and I have completed the steps in the read me linked above. A UK translation of Lady Bug Revolution was built and deployed to production (using a macOS machine). We made some changes to the instruction set as we came across issues.

An issue has created regarding production deploys (related [issue](https://github.com/phetsims/decaf/issues/126)).

Also, this build and deploy process only works on macOS platforms. We will need additional time to add cross-platform support for devs.

**Comment by Denz1994** [comment]
This issue is on-hold until a decision is made in https://github.com/phetsims/decaf/issues/127.

Once that is decided it would be nice to have another developer take these steps for a test drive.

**Comment by samreid** [comment]
We generalized the build and deploy tools and confirmed they work well for @Denz1994 and me.  In https://github.com/phetsims/decaf/issues/127 it was determined that these tools will only be supported on Mac.  @Denz1994 and I recommend having another developer follow the directions to set up their workspace for CheerpJ development, and test out by building and deploying a dev version of a sim.  @ariel-phet does this sound reasonable?  If so, can you recommend someone for that step?

**Comment by ariel-phet** [comment]
@chrisklus is a man with a mac, and is happy to try this procedure out.

**Comment by chrisklus** [comment]
I was having success following the instructions until I got an error when trying to build a sim: 

```
"Error: Could not find or load main class edu.colorado.phet.buildtools.BuildScript\n"
```

Yesterday, @Denz1994 and I attempted to determine the cause of my error, but were not able to find any differences between his state and mine that might explain the failure. Next steps are for us to pair with @samreid for some assistance debugging.

I also recommend the following changes to the instructions:

* Under "Set up SVN repo for Java development", I was able to condense steps 1, 3, and 4 into one terminal command (seen after step 4): `svn checkout https://phet.unfuddle.com/svn/phet_svn/branches/decaf ~/phet-svn-trunk-2020`. I don't think opening an IntelliJ project is necessary.
* Step 2 from that list could be promoted to its own section, just like the "Install CheerpJ" section.
* I think `gitRoot` and `urlRoot` in the build-local.json could have more specific names - those are very normal and regularly used paths that devs would see in the future and might forget that they're only used for CheepJ-related things.

**Comment by samreid** [comment]
@chrisklus and @Denz1994 and I discussed this today, and determined that one of the steps requires the developer to install IntelliJ IDEA and run one build step from within the IDE.  This is not ideal, since many developers do not have IntelliJ IDEA installed.  So we discussed these 3 possibilities:

1. Require developers to download install and use IntelliJ IDEA
2. Have someone build the classfiles and check them in somewhere
  a. check them in as a JAR file instead of scattered .class files
3. Find or develop a IDE-agnostic script or tool for building BuildScript

I recommended spending up to an hour looking for a solution for (3), since (2) requires checking in build artifacts which is usually contraindicated.  If (3) is too time consuming, we can check in something for (2).  Alternatively, we could require developers to have IntelliJ IDEA installed.

**Comment by ariel-phet** [comment]
@samreid since we are likely to have CheerpJ sims around for quite some time, (3) seems worthy of some reasonable investigation. I don't think you need to limit yourself to just 1 hour, but let's discuss if it looks like it will be more than 4-5 hours of work. Sometime in the next couple weeks sounds fine, since this is a longterm goal but not an immediate priority

**Comment by samreid** [comment]
I added instructions for compiling the BuildScript without using IntelliJ.  @Denz1994 can you please test out the instructions and the new version of buildDecaf?  If that works out OK, please send back to @chrisklus to continue testing.

**Comment by Denz1994** [comment]
Some changes were applied to the documentation to explicitly set the `JAVA_HOME` of the JDK in use. Also the logic to handle this was removed in the svn decaf branch.

The current instruction set has been tested and a successful build was pushed to bayes (found [here](https://phet-dev.colorado.edu/decaf/rotation/1.0.0-dev.11/rotation.html?simulation=rotation)).

@samreid and I advise @chrisklus to try the instruction set to deploy a dev version.

**Comment by chrisklus** [comment]
I just successfully deployed a dev version of moving-man [here](https://phet-dev.colorado.edu/decaf/moving-man/1.0.0-dev.16/moving-man.html). The process seems to be working great!

A couple suggestions:

When I tried step 3 after step 1, I got the following error:
```
svn: E155000: '/Users/chk/phet-svn-trunk-2020' is already a working copy for a different URL
```

Deleting the trunk and just running step 3 without step 1 seemed to work fine, so perhaps having both is redundant.

After the "Deploy dev (outputs a script to deploy production)" header but before the command line steps, it would be nice to have a reminder to be on VPN.

Back to @samreid to make any doc changes and close if all set here. Thanks!
